src/OfflineWorkflowsSample/OfflineWorkflowsSample/Views/Controls/MapWithTools.xaml.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Linq;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+using Esri.ArcGISRuntime.Mapping;
+using OfflineWorkflowSample.ViewModels;
+
+namespace OfflineWorkflowSample
+{
+    public sealed partial class MapWithTools : UserControl
+    {
+        public MapWithTools()
+        {
+            InitializeComponent();
+        }
+
+        public static readonly DependencyProperty MapProperty =
+            DependencyProperty.Register(
+                nameof(Map), typeof(Map),
+                typeof(MapWithTools), null
+            );
+
+        public Map Map
+        {
+            get => (Map)GetValue(MapProperty);
+            set => SetValue(MapProperty, value);
+        }
+
+        private async void Compass_Tapped(object sender, TappedRoutedEventArgs e)
+        {
+            // When the compass is tapped, reset map rotation.
+            await MyMapView.SetViewpointRotationAsync(0);
```

Comment:
This isn't needed if you bind the Compass to the MapView, instead of binding the rotation.
https://github.com/Esri/arcgis-toolkit-dotnet/blob/master/src/Esri.ArcGISRuntime.Toolkit/Common/UI/Controls/Compass/Compass.Windows.cs#L59